agent/agent.go

```diff
@@ -796,16 +796,13 @@ func (c *configInternal) APIInfo() (*api.Info, bool) {
 		// loopback, and when/if this changes localhost should resolve
 		// to IPv6 loopback in any case (lp:1644009). Review.
 		localAPIAddr := net.JoinHostPort("localhost", strconv.Itoa(port))
-		addrInAddrs := false
+		newAddrs := []string{localAPIAddr}
```

Comment:
this means non-controller agents will try to dial locally... doesn't seem ideal. put back the `addrInAddrs` check?